daemon/libnetwork/cnmallocator/networkallocator.go

```diff
@@ -949,7 +949,7 @@ func (na *cnmNetworkAllocator) allocatePools(n *api.Network) (map[netip.Prefix]s
 			}
 		}
 
-		if ic.Subnet == "" {
+		if alloc.Pool.String() != "" {
```

Comment:
This is necessary to ensure inspected network data reflects the actual allocated subnet when created with `--subnet 0.0.0.0/24` or `--subnet /24` style requests.

Reply:
Added a comment explaining the reasoning behind updating `ic.Subnet` if `alloc.Pool` is set.